### PR TITLE
Rename extensions prometheus enabled flag to extensions enabled

### PR DESCRIPTION
**Pull request by wepudt**
[K8S-11234](https://dt-rnd.atlassian.net/browse/K8S-11234)

## Description

Renames the CRD field from `.spec.extensions.prometheus.enabled` to `.spec.extensions.enabled`

## How can this be tested?

* run unit tests
* deploy with updated DK:
  ```
  apiVersion: dynatrace.com/v1beta3
  kind: DynaKube
  metadata:
    name: dynakube
    namespace: dynatrace
  spec:
    apiUrl: https://abc12345.dev.dynatracelabs.com/api
  customPullSecret: azurecr
  extensions:
      enabled: true
  templates:
    extensionExecutionController:
      customExtensionCertificates: custom-tls-cert
      imageRef:
        repository: extk8sregistry.azurecr.io/eec/dynatrace-eec
        tag: 1.301.0.20240826-125923
  ```

### CODE REVIEW

**Comment by 0sewa0** [comment]
nitpicks:
https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/eec/reconciler_test.go#L118
https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/otel/statefulset_test.go#L62
https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/reconciler_test.go#L125
https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/reconciler_test.go#L150

All of them are just test titles, where they used to be testing the `.spec.extensions.prometheus.enable=true/false`, we should also update the test titles, to remain consistent

**Comment by wepudt** [comment]
> nitpicks:
> 
> https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/eec/reconciler_test.go#L118
> 
> 
> https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/otel/statefulset_test.go#L62
> 
> 
> https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/reconciler_test.go#L125
> 
> 
> https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/controllers/dynakube/extension/reconciler_test.go#L150
> 
> All of them are just test titles, where they used to be testing the `.spec.extensions.prometheus.enable=true/false`, we should also update the test titles, to remain consistent

fixed in cfda026c0fba202206c9cbe6292189029590128a

**Review by waodim — DISMISSED** [review]
lgtm and worked for me

**Review by 0sewa0 — DISMISSED** [review]

**Comment by andriisoldatenko** [comment]
Interesting fact: if you upgrade operator from main (assuming you had already installed dk with promEnabled: true) it will gone, but dk still keep old CR. 

And if you re-apply new dk -> it will appear again.

**Review by andriisoldatenko — DISMISSED** [review]

**Comment by luhi-DT** [comment]
Make sure to update the docs if this is released! https://docs.dynatrace.com/docs/setup-and-configuration/setup-on-k8s/reference/dynakube-parameters

File: config/crd/bases/dynatrace.com_dynakubes.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.16.2
+    controller-gen.kubebuilder.io/version: v0.16.3
```
Comment (andriisoldatenko) [inline config/crd/bases/dynatrace.com_dynakubes.yaml:6]: suggested to commit also edge connect CRD
```
dynatrace-operator git:(refactor/crd_rename_prometheus_flag) ✗ gs
On branch refactor/crd_rename_prometheus_flag
Your branch is up to date with 'origin/refactor/crd_rename_prometheus_flag'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   config/crd/bases/dynatrace.com_edgeconnects.yaml
```

Comment (wepudt) [reply config/crd/bases/dynatrace.com_dynakubes.yaml:6]: fixed in d9092afdaf3ecc66dc0ec59f1b9ef2717173a97c

**Review by andriisoldatenko — APPROVED** [review]

**Review by 0sewa0 — APPROVED** [review]

**Review by waodim — APPROVED** [review]

**Pull request merged**